privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardHybridViewModel.kt

```diff
@@ -48,37 +48,72 @@ import javax.inject.Inject
 @ContributesViewModel(ActivityScope::class)
 class PrivacyDashboardHybridViewModel @Inject constructor(
     private val userWhitelistDao: UserWhitelistDao,
-    private val contentBlocking: ContentBlocking,
     private val pixel: Pixel,
     private val dispatcher: DispatcherProvider,
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
-    private val siteProtectionsViewStateMapper: SiteProtectionsViewStateMapper
+    private val siteViewStateMapper: SiteViewStateMapper,
+    private val requestDataViewStateMapper: RequestDataViewStateMapper,
+    private val protectionStatusViewStateMapper: ProtectionStatusViewStateMapper,
+    private val privacyDashboardPayloadAdapter: PrivacyDashboardPayloadAdapter
 ) : ViewModel() {
 
     private val command = Channel<Command>(1, DROP_OLDEST)
 
     sealed class Command {
         class LaunchReportBrokenSite(val data: BrokenSiteData) : Command()
+        class OpenURL(val url: String) : Command()
     }
 
     data class ViewState(
-        val userSettingsViewState: UserSettingsViewState,
-        val siteProtectionsViewState: SiteProtectionsViewState,
-        val userChangedValues: Boolean = false
+        val siteViewState: SiteViewState,
+        val userChangedValues: Boolean = false,
+        val requestData: RequestDataViewState,
+        val protectionStatus: ProtectionStatusViewState
     )
 
-    data class UserSettingsViewState(
-        val privacyProtectionEnabled: Boolean
+    data class ProtectionStatusViewState(
+        val allowlisted: Boolean,
+        val denylisted: Boolean,
+        val enabledFeatures: List<String>,
+        val unprotectedTemporary: Boolean
+    )
+
+    data class RequestDataViewState(
+        val installedSurrogates: List<String>? = null,
+        val requests: List<DetectedRequest>
+    )
+
+    data class DetectedRequest(
+        val category: String?,
+        val eTLDplus1: String?,
+        val entityName: String,
+        val ownerName: String?,
+        val pageUrl: String,
+        val prevalence: Double,
+        val state: State,
+        val url: String
     )
 
-    data class SiteProtectionsViewState(
+    sealed class State {
+        data class Blocked(val blocked: Any = Any()) : State()
```

Comment:
Why Any? Can be more specific here?

Reply:
😅 no, we can't. We need a placeholder value there so generated json is the expected one:
```
"state": {
        "blocked": {}
      }
```
tried a few things, but if you put `null` or empty values, Moshi discards adding the object. That's why ended up with that patch. Not the best thing tho, I agree.